src/pages/groupComparison/AlterationEnrichmentTypeSelector.tsx

```diff
@@ -0,0 +1,703 @@
+    ) {
+        let selectedMutations = _.keys(_.pickBy(this.currentSelectedMutations));
+        let selectedCopyNumber = _.keys(
+            _.pickBy(this.currentSelectedCopyNumber)
+        );
+        this.props.handlers.updateSelectedMutations(
+            selectedMutations as MutationEnrichmentEventType[]
+        );
+        this.props.handlers.updateSelectedCopyNumber(
+            selectedCopyNumber as CopyNumberEnrichmentEventType[]
+        );
+    }
+
+    render() {
+        return (
+            <div
+                style={{
+                    position: 'absolute',
+                    zIndex: 2,
+                    background: '#eee',
+                    borderRadius: '4px',
+                    padding: '10px',
+                }}
+            >
+                <h5>Select Alteration Types</h5>
+                <div className="checkbox">
+                    <label>
+                        <input
+                            data-test="CheckCopynumberAlterations"
+                            type="checkbox"
+                            value={'AllCopyNumberSelected'}
+                            checked={
+                                this.currentSelectedCopyNumber.HOMDEL &&
+                                this.currentSelectedCopyNumber.AMP
+                            }
+                            onClick={this.onInputClick}
+                        />{' '}
+                        Copy Number Alterations
+                    </label>
+                </div>
+                <div className="checkbox" style={{ marginLeft: '20px' }}>
+                    <label>
+                        <input
+                            data-test="DeepDeletion"
+                            type="checkbox"
+                            value={'HOMDEL'}
+                            checked={this.currentSelectedCopyNumber.HOMDEL}
+                            onClick={this.onInputClick}
+                        />{' '}
+                        Deep Deletion
+                    </label>
+                </div>
+                <div className="checkbox" style={{ marginLeft: '20px' }}>
+                    <label>
+                        <input
+                            data-test="Amplification"
+                            type="checkbox"
+                            value={'AMP'}
+                            checked={this.currentSelectedCopyNumber.AMP}
+                            onClick={this.onInputClick}
+                        />{' '}
+                        Amplification
+                    </label>
+                </div>
+                <div className="checkbox">
+                    <label>
+                        <input
+                            data-test="Mutations"
+                            type="checkbox"
+                            value={'AllMutationsSelected'}
+                            checked={
+                                this.currentSelectedMutations.missense &&
```

Comment:
`checked={this.allMutationsSelected}`

Reply:
Yes, updated (including sever refactoring of rest of the component).